app/src/main/java/com/example/mobiledger/presentation/home/HomeViewEntity.kt

```diff
@@ -5,22 +5,32 @@ import androidx.annotation.StringRes
 import com.example.mobiledger.domain.entities.TransactionEntity
 import com.example.mobiledger.domain.enums.TransactionType
 import com.github.mikephil.charting.data.PieEntry
+import com.google.firebase.Timestamp
 
 sealed class HomeViewItem(val viewType: HomeViewType) {
     data class HeaderDataRow(@StringRes val data: Int, val type: HomeViewType = HomeViewType.Header) : HomeViewItem(type)
     data class MonthlyDataRow(val data: MonthlyData, val type: HomeViewType = HomeViewType.MonthlyData) : HomeViewItem(type)
     data class MonthlyTotalPie(val pieEntryList: ArrayList<PieEntry>, val type: HomeViewType = HomeViewType.MonthlyTotalPieChart) :
         HomeViewItem(type)
 
+    data class TransactionListButton(val numOfTransactions: String, val type: HomeViewType = HomeViewType.TransactionList) :
+        HomeViewItem(type)
+
     data class TransactionDataRow(val data: TransactionData, val type: HomeViewType = HomeViewType.TransactionData) : HomeViewItem(type)
     object EmptyDataRow : HomeViewItem(HomeViewType.EmptyData)
 }
 
 data class MonthlyData(val incomeAmount: String, val expenseAmount: String)
 
 data class TransactionData(
+    val id: String,
```

Comment:
why we are sending duplicate data? As in transaction entity already had id, name etc

Reply:
Removed, It was added later when I merged your code.